File: test-suite-chatmodel-tck/src/main/java/io/micronaut/langchain4j/chatmodels/tck/tests/Friend.java
package io.micronaut.langchain4j.chatmodels.tck.tests;

import dev.langchain4j.service.SystemMessage;
import io.micronaut.context.annotation.Requires;
import io.micronaut.langchain4j.annotation.AiService;

@Requires(property = "spec.name", value = "AiServiceTest")
@AiService
public interface Friend {
    @SystemMessage("You are a good friend of mine. Answer using slang.")
    String chat(String userMessage);
}

File: test-suite-python/src/main/python/example/micronaut/aiservice/Friend.py
from abc import ABC, abstractmethod

from dev.langchain4j.service import SystemMessage
from micronaut.langchain4j.annotation import AiService


@AiService  # <1>
class Friend(ABC):

    @SystemMessage("You are a good friend of mine. Answer using slang.")  # <2>
    @abstractmethod
    def chat(self, user_message: str) -> str:
        ...
